Doc/library/stdtypes.rst

```diff
@@ -4415,8 +4415,9 @@ The constructors for both classes work the same:
    return the type of the first operand.  For example: ``frozenset('ab') |
    set('bc')`` returns an instance of :class:`frozenset`.
 
-   The following table lists operations available for :class:`set` that do not
-   apply to immutable instances of :class:`frozenset`:
+   The following mutating operations are available for :class:`set` instances.
+   For :class:`frozenset` instances, only the in-place variants of these operations
+   (``|=``, ``&=``, ``-=``, and ``^=``) are supported.
```

Comment:
Maybe it's better to use something like "augmented assignments" instead.  They actually aren't in-place ops, as frozenset is immutable.

Reply:
Thanks, I changed this in [a888e53](https://github.com/python/cpython/pull/127312/commits/a888e53072273cd4d04596d18c8e78fc59ed62ca)

Reply:
Sorry, not much better for me.  As Bénédikt explained above to you, there actually *no* variants of above methods for frozenset's.  E.g. ``|=`` for the frozenset does different things than update.

BTW, for set these "variants" actually different internally:
```pycon
$ ./python -m timeit -unsec -s 's1, s2, s3 = map(set, [[1], [2], [3]])' 's1.update(s2, s3)'
1000000 loops, best of 5: 389 nsec per loop
$ ./python -m timeit -unsec -s 's1, s2, s3 = map(set, [[1], [2], [3]])' 's1 |= s2 | s3'
200000 loops, best of 5: 1.44e+03 nsec per loop
```

Maybe we should just remove augmented "equivalents"?  From docs it's clear what's going on in update() and friends without such "aliases".  Also from [docs](https://docs.python.org/3/reference/datamodel.html#object.__ior__) - it's clear that augmented operations are defined, at least if corresponding normal operations are defined.  Which is the case both for set and frozenset.

I think that current docs are ok.  I'm going to close this, unless Raymond change his opinion.  CC @rhettinger, what do you think on just removing augmented "equivalents" for update(), etc?

Reply:
> Sorry, not much better for me. As Bénédikt explained above to you, there actually _no_ variants of above methods for frozenset's. E.g. `|=` for the frozenset does different things than update.

I get that, thank you.

> 
> BTW, for set these "variants" actually different internally:
> 
> ```
> $ ./python -m timeit -unsec -s 's1, s2, s3 = map(set, [[1], [2], [3]])' 's1.update(s2, s3)'
> 1000000 loops, best of 5: 389 nsec per loop
> $ ./python -m timeit -unsec -s 's1, s2, s3 = map(set, [[1], [2], [3]])' 's1 |= s2 | s3'
> 200000 loops, best of 5: 1.44e+03 nsec per loop
> ```
> 
> Maybe we should just remove augmented "equivalents"? From docs it's clear what's going on in update() and friends without such "aliases". 

I'm assuming you mean here to remove the augmented "equivalents" from the docs (and not the actual functionality)? If so, I think this would indeed clarify the docs.

Reply:
> remove the augmented "equivalents" from the docs

Yes.  To be precise, e.g. ``update()`` docs will look as:
```rst
   .. method:: update(*others)

      Update the set, adding elements from all others.
```

> and not the actual functionality

I'm not so evil.

> I think this would indeed clarify the docs.

I hope so.  But it seems, this is a second bugreport for years.  So, current docs probably are ok.

Reply:
> > remove the augmented "equivalents" from the docs
> 
> Yes. To be precise, e.g. `update()` docs will look as:
> 
> ```
>    .. method:: update(*others)
> 
>       Update the set, adding elements from all others.
> ```

👍 

> > and not the actual functionality
> 
> I'm not so evil.

Sure, I just wanted to clarify.

> 
> > I think this would indeed clarify the docs.
> 
> I hope so. But it seems, this is a second bugreport for years. So, current docs probably are ok.

Maybe it is clear, but I suspect it is not. As I indicated in my other comment, I think that many people will read the current docs and simply conclude that `|=` etc. does not work, without ever trying it or connecting the dots with other parts of the documentation. I think it is not unreasonable to think that from the current docs, frozenset does something to "disable" `|=`.